interfaces/builtin/serial_port.go

```diff
@@ -154,7 +156,7 @@ func (iface *SerialPortInterface) ConnectedPlugSnippet(plug *interfaces.Plug, sl
 		if iface.hasUsbAttrs(slot) {
 			// This apparmor rule must match serialDeviceNodePattern
 			// UDev tagging and device cgroups will restrict down to the specific device
-			return []byte("/dev/tty* rw,\n"), nil
+			return []byte("/dev/tty[A-Z]*[0-9] rw,\n"), nil
```

Comment:
This is why in my previous comment I said to use this:
```
// This apparmor rule is an approximation of serialDeviceNodePattern
// UDev tagging and device cgroups will restrict down to the specific device
return []byte("/dev/tty[A-Z]*[0-9] rw,\n"), nil
```

you changed the rule but not the comment discussing the rule. To specifically answer your question, AARE is not regex so in this cause they will not match perfectly, thus the 'approximation' comment.